gems/quilt_rails/README.md

```diff
@@ -119,21 +142,204 @@ function App() {
 export default App;
 ```
 
-## Rails Generators
+### Run the server
+
+`dev server`
+
+Will run the application, starting up both servers and compiling assets.
+
+## Application layout
+
+### Minimal
+
+The basic layout for an app using `quilt_rails` and friends will have a `ui` folder nested inside the normal Rails `app` folder, containing at _least_ an index.js file exporting a React component.
+
+```
+├── Gemfile (must contain "gem 'sewing_kit" and "gem 'quilt_rails'")
+├── package.json (must specify '@shopify/sewing-kit' and `@shopify/react-server` as 'dependencies')
+│
+└── app
+   └── ui
+   │   └─- index.js (exports a React component)
+   └── controllers
+       └─- react_controller (see above)
+```
+
+### Rails, Polaris, and React
+
+A more complex application will want a more complex layout. The following shows scalable locations for:
+
+- Global SCSS settings
+- App sections (roughly analogous to Rails routes)
+- Components
+- Co-located CSS modules
+- Co-located unit tests
+- Test setup files
+
+```
+└── app
+    └── ui
+        ├─- index.js (exports a React component)
+        ├── styles (optional)
+        │   └── settings.scss (global vars and @polaris overrides)
+        │
+        └── tests (optional)
+        │   └── each-test.ts
+        │   └── setup.ts
+        └── features (optional)
+            ├── App
+            │   ├── index.js
+            │   ├── App.js
+            │   └── tests
+            │       └── App.test.js
+            │
+            ├-─ MyComponent
+            │   ├-─ index.js
+            │   ├-─ MyComponent.js
+            │   ├── MyComponent.scss (optional; component-scoped CSS styles, mixins, etc)
+            │   └── tests
+            │       └── MyComponent.test.js
+            │
+            └── sections (optional; container views that compose presentation components into UI blocks)
+                └── Home
+                    ├-─ index.js
+                    └── Home.js
+```
+
+## API
+
+### ReactRenderable
+
+The `ReactRenderable` mixin is intended to be used in Rails controllers, and provides only the `render_react` method. This method handles proxying to a running `@shopify/react-server`.
+
+
+```ruby
+class ReactController < ApplicationController
+  include Quilt::ReactRenderable
+
+  def index
+    render_react
+  end
+end
+```
+
+### Engine
+
+`Quilt::Engine` provides a preconfigured controller which consumes `ReactRenderable` and provides an index route which uses it.
```

Comment:
Not sure if it is totally clear that mounting the Engine removes the need to use `ReactRenderable` in a controller.